src/vs/workbench/common/resources.ts

```diff
@@ -37,19 +39,22 @@ export class ResourceContextKey implements IContextKey<URI> {
 		this._filenameKey = ResourceContextKey.Filename.bindTo(contextKeyService);
 		this._langIdKey = ResourceContextKey.LangId.bindTo(contextKeyService);
 		this._resourceKey = ResourceContextKey.Resource.bindTo(contextKeyService);
+		this._extensionKey = ResourceContextKey.Extension.bindTo(contextKeyService);
 	}
 
 	set(value: URI) {
 		this._resourceKey.set(value);
 		this._schemeKey.set(value && value.scheme);
 		this._filenameKey.set(value && basename(value.fsPath));
 		this._langIdKey.set(value && this._modeService.getModeIdByFilenameOrFirstLine(value.fsPath));
+		this._extensionKey.set(value && paths.extname(value.fsPath));
 	}
 
 	reset(): void {
```

Comment:
@jrieken. an reason why `_filenameKey` is not reseted here?

Reply:
I'd say that is bug cc @bpasero

Reply:
@jrieken introduced by you via https://github.com/Microsoft/vscode/commit/6110660a23471d10acdecd9f6834686576718411#diff-cb108a154667420fbda68f22a3d6d90c ?

Reply:
🙉

Reply:
Should I add this missing reset in this PR?